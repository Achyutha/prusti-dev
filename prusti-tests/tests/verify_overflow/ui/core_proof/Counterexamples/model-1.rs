// compile-flags: -Punsafe_core_proof=true -Pcounterexample=true

use prusti_contracts::*;


struct X {
    a: A,
    b: B,
}

#[model]
struct X {
   a: i32,
   b: i32,
}

#[derive(Copy, Clone)]
struct A {
    a: i32,
}

#[derive(Copy, Clone)]
struct B {
    b: i32,
}

#[ensures(x.model().a == x.model().b)]
fn test1(x: X) {}

#[ensures(x.model().b == y.model().a)]
fn test2(x: X, y: X) {}

fn main() {}