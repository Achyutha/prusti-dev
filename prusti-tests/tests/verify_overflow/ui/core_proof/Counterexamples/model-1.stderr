error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/model-1.rs:27:11
   |
27 | #[ensures(x.model().a == x.model().b)]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/model-1.rs:28:1
   |
28 | fn test1(x: X) {}
   | ^^^^^^^^^^^^^^^^^
note: counterexample for "x"
       value:   X_model {
            a: 1,
            b: 0,
        }
  --> $DIR/model-1.rs:28:10
   |
28 | fn test1(x: X) {}
   |          ^
note: counterexample for "result"
       value:   ()
  --> $DIR/model-1.rs:28:16
   |
28 | fn test1(x: X) {}
   |                ^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/model-1.rs:30:11
   |
30 | #[ensures(x.model().b == y.model().a)]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/model-1.rs:31:1
   |
31 | fn test2(x: X, y: X) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^
note: counterexample for "x"
       value:   X_model {
            a: -2147477795,
            b: -2147455547,
        }
  --> $DIR/model-1.rs:31:10
   |
31 | fn test2(x: X, y: X) {}
   |          ^
note: counterexample for "y"
       value:   X_model {
            a: -2147455548,
            b: 0,
        }
  --> $DIR/model-1.rs:31:16
   |
31 | fn test2(x: X, y: X) {}
   |                ^
note: counterexample for "result"
       value:   ()
  --> $DIR/model-1.rs:31:22
   |
31 | fn test2(x: X, y: X) {}
   |                      ^^

error: aborting due to 2 previous errors

